src/models/lists/ApiListModel/ApiListModel.ts

```diff
@@ -0,0 +1,218 @@
+import { action, computed, makeObservable, observable } from 'mobx';
+
+import { MetaModel } from '../../MetaModel';
+
+import { LIMIT_PER_REQUEST_DEFAULT } from './config';
+import {
+  ApiListModelPrivateFields,
+  ApiListFetchFunction,
+  ApiListModelProps,
+  IApiListModel,
+  ResponseApiType,
+  ApiListFetchProps,
+} from './types';
+
+/**
+ * ApiListModel<T, RestApiT>
+ *
+ * @description
+ *  Модель предназначена для работы со списками, которые загружаются с сервера.
+ *  Предоставляет методы для загрузки данных с сервера с оффсетом или пагинацией, контроля состояния загрузки, сброса данных.
+ *  При достижении конца списка, модель устанавливает флаг, что список загружен.
+ *
+ * @implements {ApiListFetchProps<T, RestApiT>}
+ *  T - Тип данных, которые будут храниться в модели.
+ *  RestApiT - Тип данных, которые приходят с сервера, в случае, если нужно вернуть какие-то данные, которые не относятся к списку. По-умолчанию undefined.
+ *
+ * @param {ApiListFetchFunction<T, RestApiT>} fetchFunction - Callback-функция для загрузки данных с сервера. При вызове функции, в неё передаётся объект fetchProps: ApiListFetchProps.
+ * @param {number} [limitPerRequest] - Количество элементов, которое нужно загрузить с сервера за один запрос.
+ *
+ * @method
+ * @name load - Загрузить данные с сервера.
+ *
+ * @method
+ * @name refresh - Сбросить данные и загрузить заново.
+ *
+ * @method
+ * @name reset - Сбросить данные.
+ *
+ * @getter {list} - Список данных с типом <T>.
+ * @getter {listLoaded} - Флаг, что список загружен полностью.
+ * @getter {meta} - Мета-данные с асинхронными состояниями загрузки данных.
+ * @getter {fetchProps} - Свойства, которые передаются в функцию fetchFunction.
+ *
+ * @example
+ * import { ApiListModel } from 'mediaproject-stores';
+ * import { ArtcileType } from './types';
+ *
+ * const articlesListModel = new ApiListModel<ArtcileType>({
+ *  fetchFunction: fetchArticles,
+ *  limitPerRequest: 10,
+ * });
+ *
+ * const fetchArticles: ApiListFetchFunction<ArtcileType> = async (fetchProps) => {
+ *   const { listLength, limitCountPerRequest } = fetchProps;
+ *   const response = await fetch(`/api/list?offset=${listLength}&limit=${limitCountPerRequest}`);
+ *   const apiList = await response.json();
+ *   return { list: apiList.map(normalizer) };
+ * };
+ */
+class ApiListModel<T, RestApiT = undefined>
+  implements IApiListModel<T, RestApiT>
+{
+  private _list: T[] = [];
+
+  private _fetchFunction: ApiListFetchFunction<T, RestApiT>;
+
+  private _listLoaded = false;
+
+  private readonly _limitCountPerRequest: number;
+
+  private _meta: MetaModel = new MetaModel();
+
+  constructor({
+    fetchFunction,
+    limitPerRequest = LIMIT_PER_REQUEST_DEFAULT,
+  }: ApiListModelProps<T, RestApiT>) {
+    this._fetchFunction = fetchFunction;
+    this._limitCountPerRequest = limitPerRequest;
+
+    makeObservable<ApiListModel<T, RestApiT>, ApiListModelPrivateFields>(this, {
```

Comment:
можно сократить немного, будет удобно в случае переименования класса или его дженерик параметров
```suggestion
    makeObservable<this, ApiListModelPrivateFields>(this, {
```